src/app/util/ContentApp.cpp

```diff
@@ -0,0 +1,164 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file Contains shell commands for a commissionee (eg. end device) related to commissioning.
+ */
+
+#include <app-common/zap-generated/attribute-id.h>
+#include <app-common/zap-generated/cluster-id.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/util/ContentAppPlatform.h>
+#include <cstdio>
+#include <inttypes.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/core/DataModelTypes.h>
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/ZclString.h>
+#include <platform/CHIPDeviceLayer.h>
+
+using namespace chip;
+using namespace chip::AppPlatform;
+
+namespace chip {
+namespace AppPlatform {
+
+#define ZCL_DESCRIPTOR_CLUSTER_REVISION (1u)
+#define ZCL_APPLICATION_BASIC_CLUSTER_REVISION (1u)
+
+EmberAfStatus ContentApp::HandleReadAttribute(ClusterId clusterId, chip::AttributeId attributeId, uint8_t * buffer,
+                                              uint16_t maxReadLength)
+{
+    ChipLogProgress(DeviceLayer, "Read Attribute for device %s cluster %d attribute=%d)",
+                    GetApplicationBasic()->GetApplicationName(), static_cast<uint16_t>(clusterId),
+                    static_cast<uint16_t>(attributeId));
+
+    EmberAfStatus ret = EMBER_ZCL_STATUS_FAILURE;
+    if (clusterId == ZCL_APPLICATION_BASIC_CLUSTER_ID)
+    {
+        ret = GetApplicationBasic()->HandleReadAttribute(attributeId, buffer, maxReadLength);
+    }
+    if (clusterId == ZCL_ACCOUNT_LOGIN_CLUSTER_ID)
+    {
+        ret = GetAccountLogin()->HandleReadAttribute(attributeId, buffer, maxReadLength);
+    }
+    return ret;
+}
+
+EmberAfStatus ContentApp::HandleWriteAttribute(ClusterId clusterId, chip::AttributeId attributeId, uint8_t * buffer)
+{
+    ChipLogProgress(DeviceLayer, "Read Attribute for device %s cluster %d attribute=%d)",
+                    GetApplicationBasic()->GetApplicationName(), static_cast<uint16_t>(clusterId),
+                    static_cast<uint16_t>(attributeId));
```

Comment:
Why the lossy casting?

Reply:
same response as above - creating special log method for printing uint32.

Reply:
Addressed in https://github.com/project-chip/connectedhomeip/pull/12969